physics/MP/NSSL/mp_nssl.F90

```diff
@@ -50,16 +56,32 @@ subroutine mp_nssl_init(ncol, nlev, errflg, errmsg, threads, restart, &
 
          integer,                   intent(in)    :: mpirank
          integer,                   intent(in)    :: mpiroot
+         integer,                   intent(in)    :: mpicomm
          integer,                   intent(in)    :: imp_physics
          integer,                   intent(in)    :: imp_physics_nssl
          real(kind_phys),           intent(in)    :: nssl_cccn, nssl_alphah, nssl_alphahl
          real(kind_phys),           intent(in)    :: nssl_alphar, nssl_ehw0, nssl_ehlw0
          logical,                   intent(in)    :: nssl_ccn_on, nssl_hail_on, nssl_invertccn, nssl_3moment
 
+         real(kind_phys),           intent(inout) :: qc (:,:) !(1:ncol,1:nlev)
+         real(kind_phys),           intent(inout) :: qr (:,:) !(1:ncol,1:nlev)
+         real(kind_phys),           intent(inout) :: qi (:,:) !(1:ncol,1:nlev)
+         real(kind_phys),           intent(inout) :: qs (:,:) !(1:ncol,1:nlev)
+         real(kind_phys),           intent(inout) :: qh (:,:) !(1:ncol,1:nlev) graupel
+         real(kind_phys),           intent(inout) :: ccw(:,:) !(1:ncol,1:nlev)
+         real(kind_phys),           intent(inout) :: crw(:,:) !(1:ncol,1:nlev)
+         real(kind_phys),           intent(inout) :: cci(:,:) !(1:ncol,1:nlev)
+         real(kind_phys),           intent(inout) :: csw(:,:) !(1:ncol,1:nlev)
+         real(kind_phys),           intent(inout) :: chw(:,:) !(1:ncol,1:nlev) graupel number 
+         real(kind_phys),           intent(inout) :: vh (:,:) !(1:ncol,1:nlev) graupel volume 
+
          ! Local variables: dimensions used in nssl_init
          integer               :: ims,ime, jms,jme, kms,kme, nx, nz, i,k
          real :: nssl_params(20)
          integer :: ihailv,ipc
+         real, parameter :: qmin = 1.e-12
```

Comment:
Can you change "real" to "real(kind_phys)"
(I see a couple other instances in this module as well. Do you mind changing them as well?)

Reply:
done